server/src/main/java/org/elasticsearch/snapshots/SnapshotInfo.java

```diff
@@ -79,6 +84,170 @@
+                shardFailures = new ArrayList<>();
+            }
+
+            try {
+                if (parser.currentToken() == Token.START_ARRAY) {
+                    parser.nextToken();
+                }
+
+                while (parser.currentToken() != Token.END_ARRAY) {
+                    shardFailures.add(SnapshotShardFailure.fromXContent(parser));
+                    parser.nextToken();
+                }
+            } catch (IOException exception) {
+                throw new UncheckedIOException(exception);
+            }
+        }
+
+        private void ignoreVersion(String version) {
+            // ignore extra field
+        }
+
+        private void ignoreStartTime(String startTime) {
+            // ignore extra field
+        }
+
+        private void ignoreEndTime(String endTime) {
+            // ignore extra field
+        }
+
+        private void ignoreDurationInMillis(long durationInMillis) {
+            // ignore extra field
+        }
+
+        private SnapshotInfo build() {
+            SnapshotId snapshotId = new SnapshotId(snapshotName, snapshotUUID);
+
+            if (indices == null) {
+                indices = Collections.emptyList();
+            }
+
+            SnapshotState snapshotState = state == null ? null : SnapshotState.valueOf(state);
+            Version version = this.version == -1 ? Version.CURRENT : Version.fromId(this.version);
+
+            int totalShards = shardStatsBuilder == null ? 0 : shardStatsBuilder.getTotalShards();
+            int successfulShards = shardStatsBuilder == null ? 0 : shardStatsBuilder.getSuccessfulShards();
+
+            if (shardFailures == null) {
+                shardFailures = new ArrayList<>();
+            }
+
+            return new SnapshotInfo(snapshotId, indices, snapshotState, reason, version, startTime, endTime,
+                    totalShards, successfulShards, shardFailures, includeGlobalState);
+        }
+    }
+
+    private static final class ShardStatsBuilder {
+        private int totalShards;
+        private int successfulShards;
+
+        private void setTotalShards(int totalShards) {
+            this.totalShards = totalShards;
+        }
+
+        int getTotalShards() {
+            return totalShards;
+        }
+
+        private void setSuccessfulShards(int successfulShards) {
+            this.successfulShards = successfulShards;
+        }
+
+        int getSuccessfulShards() {
+            return successfulShards;
+        }
+
+        private void ignoreFailedShards(int failedShards) {
+            // ignore extra field
+        }
+    }
+
+    private static final ObjectParser<SnapshotInfoBuilder, Void> SNAPSHOT_INFO_PARSER =
+            new ObjectParser<>(SnapshotInfoBuilder.class.getName(), SnapshotInfoBuilder::new);
+
+    private static final ObjectParser<ShardStatsBuilder, Void> SHARD_STATS_PARSER =
+        new ObjectParser<>(ShardStatsBuilder.class.getName(), ShardStatsBuilder::new);
```

Comment:
these objects parser should be made to ignore unknown fields

Reply:
@tbrooks8 Thanks for doing this already!